relayer/src/foreign_client.rs

```diff
@@ -406,6 +476,231 @@ impl ForeignClient {
 
         Ok(())
     }
+
+    pub fn update_client_event(
+        &self,
+        consensus_height: Height,
+    ) -> Result<Option<UpdateClient>, ForeignClientError> {
+        let request = QueryClientEventRequest {
+            height: Height::zero(),
+            event_id: IbcEventType::UpdateClient,
+            client_id: self.id.clone(),
+            consensus_height,
+        };
+
+        // Tx query fails if we don't wait a bit here (block is not ready/ indexed?)
+        thread::sleep(Duration::from_millis(100));
+        let res = self
+            .dst_chain
+            .query_txs(QueryTxRequest::Client(request))
+            .map_err(|e| {
+                ForeignClientError::Misbehaviour(format!(
+                    "failed to query Tx-es for update client event {}",
+                    e
+                ))
+            })?;
+
+        if res.is_empty() {
+            return Ok(None);
+        };
+
+        assert_eq!(res.len(), 1);
```

Comment:
Great question! This was true when only one update client at a given height was allowed. In theory it should not be the case anymore. However, I just tried to do multiple updates at same height and I still see a single event. I changed the code for `query_txs()` to return them "all" and let the caller just pick one. 
Since the query is for a given client and consensus state height all the updates must have the same content so for misbehavior we just pick the first.